File: 5_yolo_python_custom_msg/src/sub.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from yolo_python_custom_msg.msg import Vision

class VisionSubscriber(Node):
    def __init__(self):
        super().__init__('vision_sub_node')

        # Membuat subscriber ke topic 'vision_topic' dengan pesan Vision
        self.subscription = self.create_subscription(
            Vision,
            'vision_topic',
            self.listener_callback,
            10)
        self.subscription  # prevent unused variable warning

    def listener_callback(self, msg):
        self.get_logger().info(f"Bounding Box : {msg.bounding_box}")
        self.get_logger().info(f"Titik Tengah : {msg.titik_tengah}")
        self.get_logger().info(f"FPS         : {msg.fps}")
        self.get_logger().info('-----------------------------')

def main(args=None):
    rclpy.init(args=args)
    node = VisionSubscriber()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: 6_yolo_cpp_custom_msg/src/sub.cpp
#include <chrono>
#include <memory>
#include <iostream>
#include "rclcpp/rclcpp.hpp"
#include "yolo_cpp_custom_msg/msg/vision.hpp"

using std::placeholders::_1;

int bounding_box = 0;
int titik_tengah = 0;
int fps = 0;

void vision_callback(const yolo_cpp_custom_msg::msg::Vision::SharedPtr msg)
{
  bounding_box = msg->bounding_box;
  titik_tengah = msg->titik_tengah;
  fps = msg->fps;

  std::cout << "Bounding Box: " << bounding_box << std::endl;
  std::cout << "Titik Tengah: " << titik_tengah << std::endl;
  std::cout << "FPS         : " << fps << std::endl;
  std::cout << "-----------------------------" << std::endl;
}

int main(int argc, char *argv[])
{
  rclcpp::init(argc, argv);
  auto node = rclcpp::Node::make_shared("vision_sub_node");
  auto subscription = node->create_subscription<yolo_cpp_custom_msg::msg::Vision>(
    "vision_topic", 10, vision_callback);

  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
